sending-a-simple-message-to-a-service/README.md

```diff
@@ -0,0 +1,325 @@
+
+Import the 
+- `ballerina/http` module to develop the REST API and define the client that can be used to send requests to the backend service
+- `ballerina/log` module to log some information for each client request
+
+```ballerina
+import ballerina/http;
+import ballerina/log;
+```
+
+3. Define two [configurable variables](https://ballerina.io/learn/by-example/#configurability) for the port the listener should listen on and the URL of the backend service.
+
+```ballerina
+configurable int port = 8290;
+configurable string healthcareBackend = "http://localhost:9090/healthcare";
+```
+
+4. Define an [`http:Client` client](https://ballerina.io/learn/by-example/#http-client) to send requests to the backend service.
+
+```ballerina
+final http:Client queryDoctorEP = check initializeHttpClient();
+
+function initializeHttpClient() returns http:Client|error => new (healthcareBackend);
+```
+
+The argument to the `new` expression is the URL for the backend service.
+
+5. Define a record corresponding to the payload from the backend service.
+
+```ballerina
+type Doctor record {|
+    string name;
+    string hospital;
+    string category;
+    string availability;
+    decimal fee;
+|};
+```
+
+The payload will be an array of JSON objects, where the structure of each JSON object matches this record. Note that you can use the "Paste JSON as record" VSCode command to generate the record if you have the JSON payload.
+
+6. Define the [HTTP service (REST API)](https://ballerina.io/learn/by-example/#rest-service) that has the resource that accepts user requests, retrieves relevant details from the backend service, and responds to the request. Use `/healthcare` as the service path (or the context) of the service which is attached to the listener listening on port `port`. Define an HTTP resource that allows the `GET` operation on resource path `/querydoctor` and accepts the `category` (corresponding to the specialization) as a path parameter.
+
+```ballerina
+service /healthcare on new http:Listener(port) {
+    resource function get querydoctor/[string category]() {
+        
+    }
+}
+```
+
+7. Implement the logic to retrieve and respond with relevant details.
+
+```ballerina
+service /healthcare on new http:Listener(port) {
+    resource function get querydoctor/[string category]() 
+            returns Doctor[]|http:NotFound|http:InternalServerError {
+        log:printInfo("Retrieving information", specialization = category);
+        
+        Doctor[]|http:ClientError resp = queryDoctorEP->/[category];
+        if resp is Doctor[] {
+            return resp;
+        }
+
+        if resp is http:ClientRequestError {
+            return <http:NotFound> {body: string `category not found: ${category}`};
+        }
+
+        return <http:InternalServerError> {body: resp.message()};
+    }
+}
+```
+
+- The `log:printInfo` statement [logs](https://ballerina.io/learn/by-example/#log) information about the request.
```

Comment:
```suggestion
- The `log:printInfo` statement [logs](https://ballerina.io/learn/by-example/#log) information about the request.
```
The flow seems to be broken here. We might need to add a sentence before these bullet points to describe what they are about.

Reply:
These are bullet points under 7.

Reply:
Then, let's make sure these points are indented properly under step 7.

Reply:
Looks OK to me - https://github.com/ballerina-guides/integration-tutorials/blob/159d69794d448e76093f38f635855c68f066ebec/sending-a-simple-message-to-a-service/README.md#step-2-develop-the-service. Checked rendering on the website also.